File: project-euler/problems/Problem12.java
package problems;

import utils.PrimeNumbers;

public class Problem12 {
	static int MAX = 1000000;

	public static void main(String args[]) {
		long sum = 0;
		long startTime = System.currentTimeMillis();
		for (int i = 1; i <= MAX; i++) {
			sum += i;
			if (PrimeNumbers.countDivisors(sum) > 500) {
				System.out.println("DONE number: " + sum + " divisors "
						+ PrimeNumbers.countDivisors(sum));
				break;
			}
		}
		long stopTime = System.currentTimeMillis();
		long elapsedTime = stopTime - startTime;
		System.out.println("time is " + elapsedTime);
	}
}

File: cracking-interview/chapter17/Problem12.py
def main(array, sum):
	dict = {}
	for x in array:
		dict[x] = True
	for x in array:
		if x > sum / 2:
			continue
		if sum - x in dict:
			print x,
			print sum - x

main([1,2,3,10,11,12],12)